#Ansible Module Name - ECS
#Test Case : Verify the playbook is throwing an error when user is trying to add Invalid Security Group to the INVALID ECS Instance
#Ansible Playbook created by - Jogesh N
---
- name: Add a INVALID Security Group To INVALID ECS Instance
  hosts: localhost
  connection: local
  gather_facts: False

  vars_files:
    - ../vars/test-environment.yml
    - ../vars/general-parameters.yml
    - ../../../vars/alicloud-creds.yml
  
  tasks:
   - block: 
     - name: Adding a INVALID Security Group To INVALID ECS Instance
       ecs:
        acs_access_key: '{{ aliyun_acs_access_key }}'
        acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
        region: '{{ region }}'
        zone: '{{ zone }}'
        group_id: '{{ addSgGroupecsInstanceIdInvalid }}' # INVALID Security Group
        instance_ids: '{{ addInvalidWindowsId }}'# INVALID ECS Instance Id
        sg_action: '{{ sgJoin }}'
    
       register: result
     - debug: var=result
     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no

     always:
      - assert:  
         that:
            - "result.changed ==  false"
            -  "'Error Message: The InstanceId provided does not exist in our records.' in result.msg[2]"
         msg: 'Test pass PASSED !!!
               This is a NEGATIVE playbook !!!
               FAILURE - EXPECTED
               REASON: {{ result}}'